File: alxSearch/src/FileUtils.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
public class FileUtils {
    public static List<Path> listTextFiles(Path dir, long maxSize) throws IOException {
        List<Path> files = new ArrayList<>();
        Files.walk(dir).forEach(p -> {
            try {
                if(Files.isRegularFile(p) && Files.size(p) <= maxSize && isTextFile(p)) {
                    files.add(p);
                }
            } catch(IOException e) {}
        });
        return files;
    }

    private static boolean isTextFile(Path file) throws IOException {
        try(InputStream in = Files.newInputStream(file)) {
            int b;
            while((b=in.read())!=-1){
                if(b==0) return false; // Nullbyte = Binärdatei
            }
        }
        return true;
    }
}

File: alxSearch/src/FileUtils.cpp
#include "FileUtils.h"
#include <filesystem>
#include <fstream>

namespace fs = std::filesystem;

namespace FileUtils {

bool isTextFile(const std::string& filepath) {
    std::ifstream file(filepath, std::ios::binary);
    if(!file) return false;
    char c;
    while(file.get(c)) {
        if(c == '\0') return false;
    }
    return true;
}

std::vector<std::string> listTextFiles(const std::string& directory, size_t max_size) {
    std::vector<std::string> files;
    for(auto& p: fs::recursive_directory_iterator(directory)) {
        if(!fs::is_regular_file(p)) continue;
        if(fs::file_size(p) > max_size) continue;
        if(isTextFile(p.path().string()))
            files.push_back(p.path().string());
    }
    return files;
}

}
